gengo.go

```diff
@@ -156,9 +158,18 @@ func genGoSingleFile(file string, conf *Config, flags GenFlags) (err error) {
 	if flags&GenFlagCheckOnly != 0 {
 		return nil
 	}
-	if err := out.WriteFile(autogen); err != nil {
-		return errors.NewWith(err, `out.WriteFile(autogen)`, -2, "(*gox.Package).WriteFile", out, autogen)
+
+	f, err := os.Create(autogen)
+	if err != nil {
+		return errors.NewWith(err, `Create(autogen)`, -2, "os.Create", autogen)
 	}
+	defer f.Close()
+	fmt.Fprint(f, generatedHeader)
```

Comment:
maybe it's better to add this feature to `github.com/goplus/gox`?

Reply:
Seems the positioning of `gox` is more like a general-purpose tool, which can also be used in other scenarios. while this header are specifically for `gop`. 

Refer to the [line stmt comment](https://github.com/goplus/gop/blob/fb3cd28365aab3caaa8798cf5cf034a2ecc024a7/cl/stmt.go#L49), adding such special header feels more in line with `gop's` behavior for me.

On the other hand, the current implementation does indeed appear to have too much same codes. I'm actually considering whether to refactor these three methods(`GenGoFiles`, `genGoSingleFile`, `genGoIn`), as they seem to have too many similarities.